test/functional/wallet_migration.py

```diff
@@ -1012,6 +1016,66 @@ def check_comments():
 
         wallet.unloadwallet()
 
+    def test_migrate_simple_watch_only(self):
+        self.log.info("Test migrating a watch-only p2pk script")
+        wallet = self.create_legacy_wallet("bare_p2pk", blank=True)
+        privkey = ECKey()
+        privkey.generate()
+        pubkey = privkey.get_pubkey()
+        p2pk_script = key_to_p2pk_script(pubkey.get_bytes())
+        wallet.importaddress(address=p2pk_script.hex())
+        res = wallet.migratewallet()
+        wo_wallet = self.nodes[0].get_wallet_rpc(res['watchonly_name'])
+        assert_equal(wo_wallet.listdescriptors()['descriptors'][0]['desc'], descsum_create(f'pk({pubkey.get_bytes().hex()})'))
+        # Ensure that migrating a wallet with watch-only scripts does not create a spendable wallet.
+        assert_equal('bare_p2pk_watchonly', res['wallet_name'])
+        assert "bare_p2pk" not in self.nodes[0].listwallets()
```

Comment:
In 37e3cab479080db9a3ac10f9b9b5c7f6c0e07882 "wallet: migration, don't create spendable wallet from a watch-only legacy wallet"

Should also check that the wallet does not appear in `listwalletdir`:

```
assert "bare_p2pk" not in [w["name"] for w in self.nodes[0].listwalletdir()["wallets"]]
```

This causes the test to fail. Examining the datadir after the test shows us that the wallet directory is still there, and the wallet.dat file is still intact and valid, although missing descriptor records.

The spendable wallet should be deleted if it is empty, although this needs to be done carefully to avoid deleting the backup file that is stored in the wallet directory.

Reply:
> This causes the test to fail. Examining the datadir after the test shows us that the wallet directory is still there, and the wallet.dat file is still intact and valid, although missing descriptor records.

Good catch. The wallet is actually entirely empty. There's neither address book records nor any transaction. We can safely remove it 👍🏼 .